File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0

    d = dict()
    
    for i in tangerine :
        if i in d:
            d[i] += 1
        else : d[i] = 1
    
    l = list(d.items())
    l.sort(key = lambda x : x[1], reverse = True)
    
    for i in l :
        if k <= 0 :
            break
        k -= d[i[0]]
        answer += 1
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for (int i : tangerine) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        
        List<Integer> list = new ArrayList(Arrays.asList(map.values().toArray()));
        list.sort(Collections.reverseOrder());
        
        for (int i : list) {
            if (k <= 0) {
                return answer;
            }
            
            k -= i;
            answer++;
        }
        
        return answer;
    }
}